src/librustdoc/html/markdown.rs

```diff
@@ -111,6 +111,25 @@ enum Line<'a> {
 }
 
 impl<'a> Line<'a> {
+    fn from_str(s: &'a str) -> Self {
+        let trimmed = s.trim();
+        match trimmed.strip_prefix("#") {
+            Some(tail) => match tail.strip_prefix("#") {
+                // `##text` rendered as `#text`.
+                Some(_) => Line::Shown(tail.into()),
+                None => match tail.as_bytes() {
+                    // `#` will be hidden.
+                    [] => Line::Hidden(""),
+                    // `# text` will be hidden.
+                    [b' ' | b'\t', ..] => Line::Hidden(tail),
```

Comment:
Why is tab added here?

Reply:
Also the old code would trim the space but this doesn't.